mlflow/utils/uri.py

```diff
@@ -447,3 +452,20 @@ def validate_path_is_safe(path):
         or (is_windows() and len(path) >= 2 and path[1] == ":")
     ):
         raise exc
+
+
+def validate_query_string(query):
+    query = _decode(query)
+    # Block query strings contain any traveral path (../) because they
+    # could be resolved as part of the path and allow path traversal.
+    if ".." in query:
+        raise MlflowException("Invalid query string", error_code=INVALID_PARAMETER_VALUE)
+
+
+def _decode(url):
+    # Keep decoding until the url stops changing
+    while True:
+        decoded = urllib.parse.unquote(url)
+        if decoded == url:
+            return url
+        url = decoded
```

Comment:
/ai any chance that this function goes into an infinite loop?

Reply:
/ai Can we use recursion instead?

Reply:
I feel like 10 is enough 🤔  Who would encode a path to such complex stuff?

Reply:
Makes sense, will limit it to 10:)

Reply:
If we reach 10, I think that's some malicious URI.